firedrake/external_operators/potential_evaluation/__init__.py

```diff
@@ -0,0 +1,405 @@
+"""
+User-facing functions to create layer potentials.
+
+Produces a Potential either through volumential or pytential
+"""
+from firedrake.functionspaceimpl import WithGeometry
+from firedrake.mesh import MeshGeometry
+__all__ = ("SingleLayerPotential",
+           "DoubleLayerPotential",
+           "VolumePotential",
+           "PotentialSourceAndTarget")
+
+
+def SingleLayerPotential(density, kernel, potential_src_and_tgt, **kwargs):
```

Comment:
This shouldn't be called `SingleLayerPotential`. There are way too many potentials to name them all out, and they just differ by the kernel. Can we have just a `LayerPotential`, or is there something preventing that?

Reply:
My main thinking was `SingleLayerPotential` would be the equivalent of [`sym.S`](https://documen.tician.de/pytential/symbolic.html#pytential.symbolic.primitives.S) and `DoubleLayerPotential` would be the equivalent of [`sym.D`](https://documen.tician.de/pytential/symbolic.html#pytential.symbolic.primitives.D).

We can just do `LayerPotential` if you think that's more clear. The `SingleLayerPotential` and `DoubleLayerPotential` functions are just one-line wrapper functions anyway

Reply:
Hmm, I think you're right. Somehow they looked much more substantial when I scrolled through than they actually are. I have no objections now, however I do think we should also expose the base `LayerPotential`, to cover things with a funny kernel. (A quadruple layer potential or D' will look funny wrapped in `SingleLayerPotential`.)